node/src/neighborhood/gossip_acceptor.rs

```diff
@@ -350,6 +341,28 @@ impl DebutHandler {
         }
     }
 
+    fn create_debut_gossip_response(
+        cryptde: &dyn CryptDE,
+        database: &mut NeighborhoodDatabase,
+        debut_node_key: PublicKey,
+    ) -> Gossip_0v1 {
+        let mut root_node = database.root().clone();
+        root_node.clear_half_neighbors();
+        root_node
+            .add_half_neighbor_key(debut_node_key.clone())
+            .unwrap_or_else(|e| {
+                panic!(
+                    "Couldn't add debuting {} as a half neighbor: {:?}",
+                    debut_node_key, e
+                )
+            });
+        let gnr =
+            GossipNodeRecord::from((root_node.inner.clone(), root_node.node_addr_opt(), cryptde));
```

Comment:
If you create a variable `node_addr_opt` first, you can write:

```
        let root_node_addr_opt = root_node.node_addr_opt();
        let gnr =
            GossipNodeRecord::from((root_node.inner, root_node_addr_opt, cryptde));
```

Notice that I spared the single .clone() call. ;-) Maybe not much, but if you think like this, you can optimize using just a little change. It's another way how you get better code, and all together it makes a big difference in overall.